src/main/scala/org/apache/spark/sql/kinesis/HDFSMetadataCommitter.scala

```diff
@@ -59,6 +62,21 @@ class HDFSMetadataCommitter[T <: AnyRef : ClassTag](path: String,
     fileContext.mkdir(metadataPath, FsPermission.getDirDefault, true)
   }
 
+  private val numRetries: Int = {
+    options.getOrElse("executor.metadata.hdfs.numretries", "3").toInt
```

Comment:
we can possibly add another safety check like this 
`options.get("executor.metadata.hdfs.numretries").map { str =>
    Try(str.toInt).toOption.filter(_ > 0).getOrElse {
      throw new IllegalArgumentException(
        s"Invalid value '$str' for option 'executor.metadata.hdfs.numretries', must be a positive integer")
    }
  }`